### ISSUE TITLE
General Error: 1215 Cannot add foreign key

### ISSUE THREAD

**Issue by louposk** [aimeos/aimeos-laravel#246]
**Environment**
1. Version 2018.10
2. Operating system Windows
3. PHP v7.2.10
4. MySql v5.7.23

After executing command 
`php artisan aimeos:setup --option=setup/default/demo:1 -v`
i get the error below:

` Checking table "users_address":
   Aimeos\MW\DB\Exception  : SQLSTATE[HY000]: General error: 1215 Cannot add foreign key constraint

  at C:\wamp64\www\aimeos\vendor\aimeos\aimeos-core\lib\mwlib\src\MW\DB\Statement\DBAL\Simple.php:91
    87|
    88|                 try {
    89|                         $result = $this->exec();
    90|                 } catch( \PDOException $e ) {
  > 91|                         throw new \Aimeos\MW\DB\Exception( $e->getMessage(), $e->getCode() );
    92|                 }
    93|
    94|                 return new \Aimeos\MW\DB\Result\PDO( $result );
    95|         }

  Exception trace:

  1   Aimeos\MW\DB\Statement\DBAL\Simple::execute()
      C:\wamp64\www\aimeos\vendor\aimeos\aimeos-core\lib\mwlib\src\MW\Setup\Task\Base.php:153

  2   Aimeos\MW\Setup\Task\Base::executeList("db-customer")
      C:\wamp64\www\aimeos\vendor\aimeos\aimeos-core\lib\mshoplib\setup\TablesCreateMShop.php:197

  3   Aimeos\MW\Setup\Task\TablesCreateMShop::setupSchema()
      C:\wamp64\www\aimeos\vendor\aimeos\aimeos-core\lib\mshoplib\setup\TablesCreateMShop.php:107

  4   Aimeos\MW\Setup\Task\TablesCreateMShop::migrate()
      C:\wamp64\www\aimeos\vendor\aimeos\aimeos-core\lib\mwlib\src\MW\Setup\Manager\Multiple.php:174

  5   Aimeos\MW\Setup\Manager\Multiple::migrateTasks()
      C:\wamp64\www\aimeos\vendor\aimeos\aimeos-core\lib\mwlib\src\MW\Setup\Manager\Multiple.php:170

  6   Aimeos\MW\Setup\Manager\Multiple::migrateTasks()
      C:\wamp64\www\aimeos\vendor\aimeos\aimeos-core\lib\mwlib\src\MW\Setup\Manager\Multiple.php:96

  7   Aimeos\MW\Setup\Manager\Multiple::migrate(Object(Aimeos\MW\Setup\Task\ListsAddTypeid))
      C:\wamp64\www\aimeos\vendor\aimeos\aimeos-laravel\src\Aimeos\Shop\Command\SetupCommand.php:75

  8   Aimeos\Shop\Command\SetupCommand::handle()
      C:\wamp64\www\aimeos\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:29

  9   call_user_func_array([])
      C:\wamp64\www\aimeos\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:29

  10  Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
      C:\wamp64\www\aimeos\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:87

  11  Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Object(Closure))
      C:\wamp64\www\aimeos\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:31

  12  Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), [])
      C:\wamp64\www\aimeos\vendor\laravel\framework\src\Illuminate\Container\Container.php:572

  13  Illuminate\Container\Container::call()
      C:\wamp64\www\aimeos\vendor\laravel\framework\src\Illuminate\Console\Command.php:183

  14  Illuminate\Console\Command::execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
      C:\wamp64\www\aimeos\vendor\symfony\console\Command\Command.php:255

  15  Symfony\Component\Console\Command\Command::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
      C:\wamp64\www\aimeos\vendor\laravel\framework\src\Illuminate\Console\Command.php:170

  16  Illuminate\Console\Command::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      C:\wamp64\www\aimeos\vendor\symfony\console\Application.php:886

  17  Symfony\Component\Console\Application::doRunCommand(Object(Aimeos\Shop\Command\SetupCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      C:\wamp64\www\aimeos\vendor\symfony\console\Application.php:262

  18  Symfony\Component\Console\Application::doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      C:\wamp64\www\aimeos\vendor\symfony\console\Application.php:145

  19  Symfony\Component\Console\Application::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      C:\wamp64\www\aimeos\vendor\laravel\framework\src\Illuminate\Console\Application.php:89

  20  Illuminate\Console\Application::run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      C:\wamp64\www\aimeos\vendor\laravel\framework\src\Illuminate\Foundation\Console\Kernel.php:122

  21  Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      C:\wamp64\www\aimeos\artisan:37`

In my database i have
`            'charset' => 'utf8',
            'collation' => 'utf8_bin',`

 and all tables are created with no problem.

Any idea?

**Comment by aimeos** [comment]
Are there records in `user_address` that have user IDs in their `parentid` column that are not existing any more in the `users` table?

**Comment by firestruck** [comment]
General error: 1215 Cannot add foreign key constraint
Same error, it´s there a solution?
I dont jhave any records in user_address or users.

**Comment by aimeos** [comment]
Do you have a mismatch of users.id and users_address.parentid column types?

**Comment by firestruck** [comment]
> Do you have a mismatch of users.id and users_address.parentid column types?

Yes yhe problem was thet i already have a table users (laravel migraion default)!
I having problems integrating in my laravel project, can you help me if you can, i'll make some donation for some beers :)
Thanks.

**Comment by nhakhtn** [comment]
you should go to table `users` then change the type of table from MyISAM to InnoDB, then it will be solved!
